app/jobs/inboxes/fetch_imap_emails_job.rb

```diff
@@ -9,8 +9,10 @@ def perform(channel)
     fetch_mail_for_channel(channel)
     # clearing old failures like timeouts since the mail is now successfully processed
     channel.reauthorized!
-  rescue Errno::ECONNREFUSED, Net::OpenTimeout, Net::IMAP::NoResponseError
+  rescue Errno::ECONNREFUSED, Net::OpenTimeout, Net::IMAP::NoResponseError, Errno::ECONNRESET, Errno::ENETUNREACH, Net::IMAP::ByeResponseError
     channel.authorization_error!
+  rescue EOFError => e
```

Comment:
What trigger EOFError? Do we need to notify the user?

Reply:
It seems random on the production amidst the account we observed. sometimes the server responds with EOFError, but on subsequent retries, it was working fine. 

ref: https://github.com/tpitale/mail_room/issues/15